### PR TITLE
Add ability for SQLKit drivers to expose version numbers

### PR DESCRIPTION
**Pull request by gwynne**
Also fills in some missing documentation.

### CODE REVIEW

File: Sources/SQLKit/SQLDatabase.swift [hunk 1]
```diff
@@ -1,14 +1,94 @@
...
+    /// may also change at runtime (for example, if a connection is auto-reconnected after a remote
+    /// update), or even become unknown again after being known.
+    ///
+    /// - Warning: This version number has nothing to do with ``SQLKit`` or (usually) of the driver
+    ///   implementation for the database, nor does it represent any data stored within the database;
+    ///   it is the version of the database implementation _itself_ (such as of a MySQL server or
+    ///   `libsqlite3` library). A significant part of the motivation to finally add this property comes
+    ///   from a larger desire to enable customizing a given ``SQLDialect``'s configuration based on the
+    ///   actual feature set available at runtime instead of having to hardcode a "safe" baseline.
+    var version: SQLDatabaseReportedVersion? { get }
```
Comment (0xTim) [inline Sources/SQLKit/SQLDatabase.swift:59]: Is it worth renaming this to `databaseVersion` or similar to avoid any ambiguity

**Review by 0xTim — APPROVED** [review]
LGTM - one query around naming but take it or leave it

File: Sources/SQLKit/SQLDatabase.swift (hunk 1 above)
Comment (gwynne) [reply Sources/SQLKit/SQLDatabase.swift:59]: I'm already unhappy about the repetition of "database" in the name of the type, honestly 😕

Comment (0xTim) [reply Sources/SQLKit/SQLDatabase.swift:59]: Fair - I'm just thinking of potential future versions (no pun intended) where we expose other versions, but I'm happy with this as is

**Pull request merged**